test/test_ops.py

```diff
@@ -412,14 +413,29 @@ def _ref_test_helper(self, ctx, device, dtype, op):
                     exact_layout=True,
                     exact_is_coalesced=True,
                 )
-            except AssertionError as e:
-                # Raises the error if the precise dtype comparison wouldn't be
-                # different
-                if dtype is precise_dtype:
-                    raise e
+            except AssertionError:
+                # see issue #78389:
```

Comment:
I looked at the linked issue (and added the primTorch label) but I'm not sure I understand -- 

where is `a.copy_(b)` called? And when `a` and `b` are the same dtype then `a != b` after an inplace copy?

Reply:
The `a` and `b` arguments are getting promoted in `elementwise_type_promotion_wrapper`. This promotion wrapper calls `_maybe_convert_to_dtype` that calls `prims.convert_element_type` which does the `copy`. I will prepare a code snippet to replicate the problem in the issue #78389. Apparently this is not a primTorch issue, but general pytorch